test/integration/dashboardTest/submitNewLocationTest.js

```diff
@@ -0,0 +1,508 @@
+
+// Setup chai
+chai.use(chaiHttp)
+chai.use(sinonChai)
+
+const sandbox = sinon.createSandbox()
+
+const expect = chai.expect
+
+describe('dashboard.js integration tests: submitNewpLocation', () => {
+  describe('for a request that contains valid non-empty fields', () => {
+    beforeEach(async () => {
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClient()
+
+      this.client = await clientFactory(db)
+
+      this.agent = chai.request.agent(server)
+
+      await this.agent.post('/login').send({
+        username: helpers.getEnvVar('WEB_USERNAME'),
+        password: helpers.getEnvVar('PASSWORD'),
+      })
+
+      this.locationid = 'unusedID'
+      this.displayName = 'locationName'
+      this.doorCoreId = 'door_coreID'
+      this.radarCoreId = 'radar_coreID'
+      this.radarType = 'XeThru'
+      this.responderPhoneNumber = '+18001231234'
+      this.twilioNumber = '+15005550006'
+      this.firmwareStateMachine = false
+      this.alertApiKey = 'myApiKey'
+      const goodRequest = {
+        locationid: this.locationid,
+        displayName: this.displayName,
+        doorCoreID: this.doorCoreId,
+        radarCoreID: this.radarCoreId,
+        radarType: this.radarType,
+        responderPhoneNumber: this.responderPhoneNumber,
+        twilioPhone: this.twilioNumber,
+        firmwareStateMachine: this.firmwareStateMachine.toString(),
+        alertApiKey: this.alertApiKey,
+        clientId: this.client.id,
+      }
+
+      this.response = await this.agent.post('/locations').send(goodRequest)
+    })
+
+    afterEach(async () => {
+      await db.clearSessions()
+      await db.clearLocations()
+      await db.clearClients()
+
+      this.agent.close()
+    })
+
+    it('should return 200', () => {
+      expect(this.response).to.have.status(200)
+    })
+
+    it('should create a location in the database with the given values', async () => {
+      const newLocation = await db.getLocationData(this.locationid)
+
+      expect({
+        locationid: newLocation.locationid,
+        displayName: newLocation.displayName,
+        doorCoreId: newLocation.doorCoreId,
+        radarCoreId: newLocation.radarCoreId,
+        radarType: newLocation.radarType,
+        responderPhoneNumber: newLocation.responderPhoneNumber,
+        twilioNumber: newLocation.twilioNumber,
+        firmwareStateMachine: newLocation.firmwareStateMachine,
+        alertApiKey: newLocation.alertApiKey,
+        clientId: newLocation.client.id,
+      }).to.eql({
+        locationid: this.locationid,
+        displayName: this.displayName,
+        doorCoreId: this.doorCoreId,
+        radarCoreId: this.radarCoreId,
+        radarType: this.radarType,
+        responderPhoneNumber: this.responderPhoneNumber,
+        twilioNumber: this.twilioNumber,
+        firmwareStateMachine: this.firmwareStateMachine,
+        alertApiKey: this.alertApiKey,
+        clientId: this.client.id,
```

Comment:
It wasn't checking the values that were passed were actually being put into the DB, so this `expects` is new.